tests/units/result_manager_models.py

```diff
@@ -0,0 +1,166 @@
+
+        result = TestResult(**test_definition['input'])
+
+        result.is_failure()
+        if result.result == 'failure' and len(result.messages) == len(test_definition['input']['messages']) if 'messages' in test_definition['input'] else 0:
+            logging.debug('is_failure only is working')
+
+        result.is_failure('it is a great failure')
+        if result.result == 'failure' and len(result.messages) == len(test_definition['input']['messages'])+1 if 'messages' in test_definition['input'] else 1:
+            logging.debug('is_failure with message is working')
+
+    @pytest.mark.parametrize("test_definition", TEST_RESULT_UNIT, ids=generate_test_ids_dict)
+    def test_anta_result_set_status_error(self, test_definition):
+        """Test Result model.
+
+        Test structure:
+        ---------------
+            {
+                'name': 'valid_full',
+                'input': {"host": '1.1.1.1', 'test': 'pytest_unit_test', 'result': 'success', 'messages': ['test']},
+                'expected_result': 'valid',
+            }
+
+        """
+        if test_definition['expected_result'] == 'invalid':
+            pytest.skip('Not concerned by the test')
+
+        result = TestResult(**test_definition['input'])
+
+        result.is_error()
+        if result.result == 'error' and len(result.messages) == len(test_definition['input']['messages']) if 'messages' in test_definition['input'] else 0:
+            logging.debug('is_error only is working')
+
+        result.is_error('it is a great error')
+        if result.result == 'error' and len(result.messages) == len(test_definition['input']['messages'])+1 if 'messages' in test_definition['input'] else 1:
+            logging.debug('is_error with message is working')
+
+    @pytest.mark.parametrize("test_definition", TEST_RESULT_UNIT, ids=generate_test_ids_dict)
+    def test_anta_result_set_status_skipped(self, test_definition):
+        """Test Result model.
+
+        Test structure:
+        ---------------
+            {
+                'name': 'valid_full',
+                'input': {"host": '1.1.1.1', 'test': 'pytest_unit_test', 'result': 'success', 'messages': ['test']},
+                'expected_result': 'valid',
+            }
+
+        """
+        if test_definition['expected_result'] == 'invalid':
+            pytest.skip('Not concerned by the test')
+
+        result = TestResult(**test_definition['input'])
+
+        result.is_skipped()
+        if result.result == 'skipped' and len(result.messages) == len(test_definition['input']['messages']) if 'messages' in test_definition['input'] else 0:
+            logging.debug('is_skipped only is working')
+
+        result.is_skipped('it is a great skipped')
+        if result.result == 'skipped' and len(result.messages) == len(test_definition['input']['messages'])+1 if 'messages' in test_definition['input'] else 1:
+            logging.debug('is_skipped with message is working')
```

Comment:
I would write it like this to make clear what condition is failing in each assert (rather than using `and`)
```suggestion
        assert result.result == 'skipped'
        result_message_len = len(result.messages)
        if 'messages' in test_definition['input']:
            assert  result_message_len == len(test_definition['input']['messages'])
        else:
            assert result_message_len == 0
       
        # Adding one message
        result.is_skipped('it is a great skipped')
        assert result.result == 'skipped'
        assert len(result.messages) = result_message_len + 1
```